modules/programs/hyprlauncher.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    types
+    mkIf
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    ;
+
+  cfg = config.programs.hyprlauncher;
+in
+{
+  meta.maintainers = with lib.hm.maintainers; [ aguirre-matteo ];
+
+  options.programs.hyprlauncher = {
+    enable = mkEnableOption "hyprlauncher";
+    package = mkPackageOption pkgs "hyprlauncher" { nullable = true; };
+    settings = mkOption {
+      type =
+        with lib.types;
+        let
+          valueType =
+            nullOr (oneOf [
+              bool
+              int
+              float
+              str
+              path
+              (attrsOf valueType)
+              (listOf valueType)
+            ])
+            // {
+              description = "Hyprland configuration value";
+            };
+        in
+        valueType;
+      default = { };
+      example = {
+        general.grab_focus = true;
+        cache.enabled = true;
+        ui.window_size = "400 260";
+        finders = {
+          math_prefix = "=";
+          desktop_icons = true;
+        };
+      };
+      description = ''
+        Configuration settings for hyprlauncher. All the available options can be found here:
+        <https://wiki.hypr.land/Hypr-Ecosystem/hyprlauncher/#config>
+      '';
+    };
+    launchAsDaemon = mkOption {
+      type = types.bool;
+      default = true;
+      example = false;
+      description = ''
+        Whether to launch hyprlauncher as a daemon adding it to Hyprland's exec-once list.
+      '';
+    };
```

Comment:
Only thing I was was wondering about was option name and possible future changes. If someone wanted to add a systemd service for this instead of relying on hyprland configuration through home manager.